libs/parallelism/async_base/tests/unit/launch_policy.cpp

```diff
@@ -0,0 +1,58 @@
+//  Copyright (c) 2021 Hartmut Kaiser
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/config.hpp>
+#include <hpx/hpx_main.hpp>
+#include <hpx/modules/async_base.hpp>
+#include <hpx/modules/coroutines.hpp>
+#include <hpx/modules/testing.hpp>
+
+#include <cstdint>
+
+///////////////////////////////////////////////////////////////////////////////
+template <typename Launch>
+void test_policy(Launch policy)
+{
+    HPX_TEST(policy.priority() == hpx::threads::thread_priority::default_);
+    HPX_TEST(policy.stacksize() == hpx::threads::thread_stacksize::default_);
+    HPX_TEST(
+        policy.hint().mode == hpx::threads::thread_schedule_hint_mode::none);
+    HPX_TEST_EQ(policy.hint().hint, std::int16_t(-1));
+
+    policy = hpx::threads::thread_priority::normal;
+    HPX_TEST(policy.priority() == hpx::threads::thread_priority::normal);
+
+    policy = hpx::threads::thread_stacksize::medium;
+    HPX_TEST(policy.stacksize() == hpx::threads::thread_stacksize::medium);
+
+    hpx::threads::thread_schedule_hint hint(0);
+    policy = hint;
+    HPX_TEST(policy.hint().mode == hint.mode);
+    HPX_TEST_EQ(policy.hint().hint, hint.hint);
+}
+
+int main()
+{
+    HPX_TEST_LTE(sizeof(hpx::launch::async_policy), sizeof(std::int64_t));
+    HPX_TEST_LTE(sizeof(hpx::launch::sync_policy), sizeof(std::int64_t));
+    HPX_TEST_LTE(sizeof(hpx::launch::deferred_policy), sizeof(std::int64_t));
+    HPX_TEST_LTE(sizeof(hpx::launch::fork_policy), sizeof(std::int64_t));
+    HPX_TEST_LTE(sizeof(hpx::launch::apply_policy), sizeof(std::int64_t));
```

Comment:
```suggestion
    static_assert(sizeof(hpx::launch::async_policy) == sizeof(std::int64_t));
    static_assert(sizeof(hpx::launch::sync_policy) == sizeof(std::int64_t));
    static_assert(sizeof(hpx::launch::deferred_policy) == sizeof(std::int64_t));
    static_assert(sizeof(hpx::launch::fork_policy) == sizeof(std::int64_t));
    static_assert(sizeof(hpx::launch::apply_policy) == sizeof(std::int64_t));
```

Reply:
Should we rely on `==`, I think `<=` would be more appropriate.

Reply:
Uh, brainfart. I only meant the change from `HPX_TEST_X` to `static_assert`. The change to `==` was unintentional.